### PR TITLE
Adding pop-up menu and context menu widget for loadouts

### PR DESCRIPTION
**Pull request by loganlearner**
This adds a collapsable popup menu to the context menu to quickly select between loadouts. Alongside there are minor user feedback changes such as sounds to the buttons to change and reset your loadout. 

Todo:
- [x] Add reset menu to mini-menu
- [x] Check scaling for other resolutions
- [x] Icon for context menu button

image

### CODE REVIEW

File: lua/autorun/server/sv_cfc_loadout_init.lua [hunk 1]
```diff
@@ -4,5 +4,6 @@ util.AddNetworkString( "CFC_Loadout_InitialSpawn" )
 util.AddNetworkString( "CFC_Loadout_SendRestrictions" )
 include( "cfc_loadouts/server/sv_cfc_loadout.lua" )
 AddCSLuaFile( "cfc_loadouts/client/cl_cfc_functions.lua" )
-AddCSLuaFile( "cfc_loadouts/client/cl_cfc_interface.lua" )
-AddCSLuaFile( "cfc_loadouts/client/cl_cfc_loadout.lua" )
\ No newline at end of file
+-- AddCSLuaFile( "cfc_loadouts/client/cl_cfc_interface.lua" )
```
Comment (wrefgtzweve) [inline lua/autorun/server/sv_cfc_loadout_init.lua:7]: Could just remove this if the file doesn't exist anymore

File: lua/cfc_loadouts/client/cl_cfc_loadout.lua [hunk 2]
```diff
@@ -162,17 +163,21 @@ function CFCLoadouts.openLoadout()
...
     resetSelectButton:SetSize( 144 * uiScale, 27 * uiScale )
-    resetSelectButton:SetText( "Select default loadout" )
+    resetSelectButton:SetText( "Reset to default" )
     CFCLoadouts.paintButton( resetSelectButton )
 
     resetSelectButton.DoClick = function()
         net.Start( "CFC_Loadout_Resetweapons" )
         net.SendToServer()
+
+        EmitSound( Sound( "Weapon_357.OpenLoader" ), LocalPlayer():GetPos(), 1, CHAN_AUTO, 1, 75, 0, 100 )
```
Comment (wrefgtzweve) [inline lua/cfc_loadouts/client/cl_cfc_loadout.lua:180]: Might be better to use [surface.PlaySound](https://wiki.facepunch.com/gmod/surface.PlaySound)

File: lua/cfc_loadouts/client/cl_cfc_quickmenu.lua [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    loadoutSelectButton.DoClick = function()
+        local _, line = loadoutPreviewList:GetSelectedLine()
+        if line == nil then return end
+        local selectedWeapons = CFCLoadouts.getLoadoutJsonTable( line:GetValue( 1 ) )
+        net.Start( "CFC_Loadout_WeaponTable" )
+        net.WriteTable( selectedWeapons )
+        net.SendToServer()
+
+        EmitSound( Sound( "Weapon_357.Reload" ), LocalPlayer():GetPos(), 1, CHAN_AUTO, 1, 75, 0, 100 )
```
Comment (wrefgtzweve) [inline lua/cfc_loadouts/client/cl_cfc_quickmenu.lua:76]: Same as the other EmitSound

**Review by wrefgtzweve — COMMENTED** [review]
Looks good to me, just some small nitpicks. 
Not great at vgui so I can't really give a good review on that, worked fine in game though!

And maybe add some sort of background to the icon to make it more pop out.
image

File: lua/cfc_loadouts/client/cl_cfc_loadout.lua (hunk 2 above)
Comment (loganlearner) [reply lua/cfc_loadouts/client/cl_cfc_loadout.lua:180]: Sure, I'll get to it

File: lua/autorun/server/sv_cfc_loadout_init.lua (hunk 1 above)
Comment (loganlearner) [reply lua/autorun/server/sv_cfc_loadout_init.lua:7]: Sure thing, although I will say that It never existed...

**Review by wrefgtzweve — APPROVED** [review]
Looks good to me! no errors or anything clunky. The icon can still be slightly hard to see sometimes but that's really nitpicky. Another nitpick I have is that maybe the widget should close after a while on its own, currently when the context menu is opened it will remain open till closed even if it's for example and hour later. Again though, looks all good to me and ready to merge!

**Pull request merged**